### PR TITLE
Add a queryable method to Encryptors

### PR DESCRIPTION
**Pull request by ksdev-pl**
A proposition related to: #5099

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ksdev-pl Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=5102)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=5102) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ksdev-pl Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=5102)!

File: crypto/src/main/java/org/springframework/security/crypto/encrypt/Encryptors.java [hunk 1]
```diff
@@ -90,6 +90,21 @@ public static TextEncryptor text(CharSequence password, CharSequence salt) {
...
+	 * encryption. Uses a 16-byte all-zero initialization vector so encrypting the same
+	 * data results in the same encryption result. This is done to allow encrypted data to
+	 * be queried against.
+	 *
+	 * @param password the password used to generate the encryptor's secret key; should
+	 * not be shared
+	 * @param salt a hex-encoded, random, site-global salt value to use to generate the
+	 * secret key
+	 */
+	public static BytesEncryptor queryable(CharSequence password, CharSequence salt) {
```
Comment (SangameshUppin) [inline crypto/src/main/java/org/springframework/security/crypto/encrypt/Encryptors.java]: Why is the method called 'queryable'?, Should this class allow encryptions specific to make query.

Comment (ksdev-pl) [reply crypto/src/main/java/org/springframework/security/crypto/encrypt/Encryptors.java]: I'm not sure if it's the best name. But currently we have:
- (Encryptor) which is "stronger"
- (Encryptor) which is "standard"
- (Encryptor) which is "delux"
- so an Encryptor which is "queryable" seemed not so bad.

Currently we have a method named "queryableText", which returns a `TextEncryptor`. Maybe the same method which returns a `BytesEncryptor` should be called "queryableBytes"? But we don't have "strongerBytes" or "standardBytes", just "stronger" and "standard", so maybe the "queryable" is enough?

**Comment by rwinch** [comment]
I don't want to continue adding methods to Encryptors. The class was a mistake because it assumes that security is static. What was stronger when the class was written is no longer stronger. Instead we should open up the constructors of the specific implementations and deprecate algorithms that are not secure anymore. If you would, like you can submit a new PR to make AesByesEncryptor public which should achieve the same end result for you.

**Pull request closed without merging**